benchmarks/BaselinesAsyncSocket.cpp

```diff
@@ -0,0 +1,248 @@
+  EventBase& eventBase_;
+  const size_t loadSize_;
+  const size_t recvBufferLength_;
+  size_t receivedLength_{0};
+};
+
+class ServerAcceptCallback : public AsyncServerSocket::AcceptCallback {
+ public:
+  ServerAcceptCallback(
+      EventBase& eventBase,
+      size_t loadSize,
+      size_t recvBufferLength)
+      : eventBase_(eventBase),
+        loadSize_(loadSize),
+        recvBufferLength_(recvBufferLength) {}
+
+  void connectionAccepted(
+      int fd,
+      const SocketAddress&) noexcept override {
+    auto socket =
+        folly::AsyncSocket::UniquePtr(new AsyncSocket(&eventBase_, fd));
+    new TcpReader(
+        std::move(socket), eventBase_, loadSize_, recvBufferLength_);
+  }
+
+  void acceptError(const std::exception& ex) noexcept override {
+    LOG(ERROR) << "acceptError" << ex.what() << std::endl;
+    eventBase_.terminateLoopSoon();
+  }
+
+ private:
+  EventBase& eventBase_;
+  const size_t loadSize_;
+  const size_t recvBufferLength_;
+};
+
+class TcpWriter : public ::folly::AsyncTransportWrapper::WriteCallback {
+ public:
+  void startWriting(AsyncSocket& socket, size_t loadSize, size_t messageSize) {
+    size_t bytesSent{0};
+
+    while (!closed_ && bytesSent < loadSize) {
+      auto data = IOBuf::copyBuffer(std::string(messageSize, 'a'));
+      socket.writeChain(this, std::move(data));
+      bytesSent += messageSize;
+    }
+  }
+
+ private:
+  void writeSuccess() noexcept override {
+  }
+
+  void writeErr(
+      size_t,
+      const folly::AsyncSocketException& exn) noexcept override {
+    LOG(ERROR) << "writeError: " << exn.what();
+    closed_ = true;
+  }
+
+  bool closed_{false};
+};
+
+class ClientConnectCallback : public AsyncSocket::ConnectCallback {
+ public:
+  ClientConnectCallback(EventBase& eventBase, size_t loadSize, size_t msgLength)
+      : eventBase_(eventBase), loadSize_(loadSize), msgLength_(msgLength) {}
+
+  void connect() {
+    eventBase_.runInEventBaseThread([this] {
+      socket_.reset(new AsyncSocket(&eventBase_));
+      SocketAddress clientAaddr("::", PORT);
+      socket_->connect(this, clientAaddr);
+    });
+  }
+
+ private:
+  void connectSuccess() noexcept override {
+    TcpWriter writer;
+    writer.startWriting(*socket_, loadSize_, msgLength_);
+    socket_->close();
+    delete this;
+  }
+
+  void connectErr(const AsyncSocketException& ex) noexcept override {
+    LOG(ERROR) << "connectErr: " << ex.what() << " " << ex.getType();
+    delete this;
+  }
+
+  AsyncSocket::UniquePtr socket_;
+  EventBase& eventBase_;
+  const size_t loadSize_;
+  const size_t msgLength_;
+};
+}
+
+static void BM_Baseline_AsyncSocket_SendReceive(
+    size_t loadSize,
+    size_t msgLength,
+    size_t recvLength) {
+  EventBase serverEventBase;
+  auto serverSocket = AsyncServerSocket::newSocket(&serverEventBase);
+
+  ServerAcceptCallback serverCallback(serverEventBase, loadSize, recvLength);
+
+  SocketAddress addr("::", PORT);
```

Comment:
I think we should be using port 0 instead of a hardcoded port.  We can get the list of listened-on addresses from the `AsyncServerSocket` and pass the resolved ephemeral port to the client.